pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,406 @@
+
+	cmd.Flags().UintVar(virtComponents["virt-launcher"], "virt-launcher", NoFlag, "show/set virt-launcher log verbosity (0-9)")
+	cmd.Flags().Lookup("virt-launcher").NoOptDefVal = strconv.FormatUint(noArg, 10)
+
+	cmd.Flags().UintVar(virtComponents["virt-operator"], "virt-operator", NoFlag, "show/set virt-operator log verbosity (0-9)")
+	cmd.Flags().Lookup("virt-operator").NoOptDefVal = strconv.FormatUint(noArg, 10)
+
+	cmd.Flags().UintVar(virtComponents[allComponent], allComponent, NoFlag, "show/set all component log verbosity (0-9)")
+	cmd.Flags().Lookup(allComponent).NoOptDefVal = strconv.FormatUint(noArg, 10)
+
+	cmd.Flags().BoolVar(&isReset, "reset", false, "reset log verbosity to the default verbosity (2) (empty the log verbosity)")
+
+	// cannot specify "reset" and "all" flag at the same time
+	cmd.MarkFlagsMutuallyExclusive("reset", allComponent)
+
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+// Command line flag syntax:
+//
+//	OK: --flag=x
+//	NG: --flag x
+//
+// Another caveat of NoOptDefVal is that
+// we cannot use the "--flag x" syntax, because "--flag x" only applies to flags without a default value.
+// See vendor/github.com/spf13/pflag/flag.go, especially note the order of the if clause below.
+// https://github.com/kubevirt/kubevirt/blob/main/vendor/github.com/spf13/pflag/flag.go#L983C1-L998C3
+func usage() string {
+	return `  # reset (to default) log-verbosity for all components:
+  {{ProgramName}} adm logVerbosity --reset
+
+  # show log-verbosity for all components:
+  {{ProgramName}} adm log-verbosity --all
+  # set log-verbosity to 3 for all components:
+  {{ProgramName}} adm log-verbosity --all=3
+
+  # show log-verbosity for virt-handler:
+  {{ProgramName}} adm log-verbosity --virt-handler
+  # set log-verbosity to 7 for virt-handler:
+  {{ProgramName}} adm log-verbosity --virt-handler=7
+
+  # show log-verbosity for virt-handler and virt-launcher:
+  {{ProgramName}} adm log-verbosity --virt-handler --virt-launcher
+  # set log-verbosity for virt-handler to 7 and virt-launcher to 3:
+  {{ProgramName}} adm log-verbosity --virt-handler=7 --virt-launcher=3
+
+  # reset all components to default besides virt-handler which is 7:
+  {{ProgramName}} adm log-verbosity --reset --virt-handler=7
+  # set all components to 3 besides virt-handler which is 7:
+  {{ProgramName}} adm log-verbosity --all=3 --virt-handler=7`
+}
+
+// component name to JSON name
+func getJSONNameByComponentName(componentName string) string {
+	var componentNameToJSONName = map[string]string{
+		"virt-api":        "virtAPI",
+		"virt-controller": "virtController",
+		"virt-handler":    "virtHandler",
+		"virt-launcher":   "virtLauncher",
+		"virt-operator":   "virtOperator",
+		allComponent:      allComponent,
+	}
+	return componentNameToJSONName[componentName]
+}
+
+func detectInstallNamespaceAndName(virtClient kubecli.KubevirtClient) (namespace, name string, err error) {
```

Comment:
nit: please don't use the named return values. Just add a comment in the function description with the returned values

Reply:
I used the named return values because `golangci-lint` did not like more than 2 unnamed results.
```
pkg/virtctl/adm/logverbosity/logverbosity.go:181:1: unnamedResult: consider giving a name to these results (gocritic)
func detectInstallNamespaceAndName(virtClient kubecli.KubevirtClient) (string, string, error) {
```
I tried to remove all the linter messages. However, this time I cannot find a way to remove the linter message. So I leave it as it is. I think linter is not required, so I hope it is ok that I leave the linter message.

I will remove the named return values as you suggested. I think the return values of the function are clear enough from the function name and variable names, so I will not leave any comment on the function.

Reply:
Done